SofaKernel/modules/SofaCore/src/sofa/core/objectmodel/BaseNode.cpp

```
@@ -77,8 +77,17 @@ void BaseNode::setObjectContext(BaseObject::SPtr obj)
 /// Reset the context of an object
 void BaseNode::clearObjectContext(BaseObject::SPtr obj)
 {
-    if (obj->getContext() == this->getContext())
-        obj->l_context.reset();
+    if (obj != nullptr)
+    {
+        if (obj->getContext() == this->getContext())
+        {
+            obj->l_context.reset();
+        }
+    }
+    else
+    {
+        msg_warning() << "Trying to clear a nullptr object context";
```

To me this should throw an exception... This is clearly the caller's job to check this and print an error message if it knows why it happened.